flaky test TestZkClientAsyncRetry.testAsyncRetryCategories

https://github.com/apache/helix/runs/1846999304?check_suite_focus=true

> 2021-02-07T00:51:39.8461493Z ##[group]Run cd zookeeper-api; mvn -q test
2021-02-07T00:51:39.8462196Z [36;1mcd zookeeper-api; mvn -q test[0m
2021-02-07T00:51:39.8505908Z shell: /bin/bash -e {0}
2021-02-07T00:51:39.8506301Z env:
2021-02-07T00:51:39.8506798Z   JAVA_HOME_8.0.282_x64: /opt/hostedtoolcache/jdk/8.0.282/x64
2021-02-07T00:51:39.8507421Z   JAVA_HOME: /opt/hostedtoolcache/jdk/8.0.282/x64
2021-02-07T00:51:39.8508027Z   JAVA_HOME_8_0_282_X64: /opt/hostedtoolcache/jdk/8.0.282/x64
2021-02-07T00:51:39.8508539Z ##[endgroup]
2021-02-07T00:51:45.8183981Z Starting ZK server at localhost:2127
2021-02-07T00:51:47.0066489Z Starting TestFederatedZkClient
2021-02-07T00:51:47.1377560Z 1    [ZkClient-EventThread-59-localhost:2127] ERROR org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks  - Interrupted waiting for success
2021-02-07T00:51:47.1405896Z java.lang.InterruptedException
2021-02-07T00:51:47.1406952Z 	at java.lang.Object.wait(Native Method)
2021-02-07T00:51:47.1407801Z 	at java.lang.Object.wait(Object.java:502)
2021-02-07T00:51:47.1409790Z 	at org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks$DefaultCallback.waitForSuccess(ZkAsyncCallbacks.java:220)
2021-02-07T00:51:47.1412133Z 	at org.apache.helix.zookeeper.zkclient.ZkClient.issueSync(ZkClient.java:1299)
2021-02-07T00:51:47.1413968Z 	at org.apache.helix.zookeeper.zkclient.ZkClient.access$300(ZkClient.java:85)
2021-02-07T00:51:47.1423184Z 	at org.apache.helix.zookeeper.zkclient.ZkClient$4.run(ZkClient.java:1325)
2021-02-07T00:51:47.1431421Z 	at org.apache.helix.zookeeper.zkclient.ZkEventThread.run(ZkEventThread.java:99)
2021-02-07T00:51:49.2402036Z Ending TestFederatedZkClient
2021-02-07T00:51:49.3388463Z Starting ZK server at localhost:33917
2021-02-07T00:51:51.9491019Z Starting ZK server at localhost:44937
2021-02-07T00:51:58.6741175Z 11541 [Thread-73] ERROR org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks  - Interrupted waiting for success
2021-02-07T00:51:58.6744706Z java.lang.InterruptedException
2021-02-07T00:51:58.6745604Z 	at java.lang.Object.wait(Native Method)
2021-02-07T00:51:58.6746574Z 	at java.lang.Object.wait(Object.java:502)
2021-02-07T00:51:58.6748670Z 	at org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks$DefaultCallback.waitForSuccess(ZkAsyncCallbacks.java:220)
2021-02-07T00:51:58.6753634Z 	at org.apache.helix.zookeeper.impl.client.TestZkClientAsyncRetry.lambda$waitAsyncOperation$0(TestZkClientAsyncRetry.java:65)
2021-02-07T00:51:58.6756534Z 	at java.lang.Thread.run(Thread.java:748)
2021-02-07T00:51:59.2451472Z 12112 [main] ERROR org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer  - Data size: 311979 is greater than 204800 bytes, is compressed: false, ZNRecord.id: record. Data will not be written to Zookeeper.
2021-02-07T00:51:59.2967857Z 12163 [main] ERROR org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer  - Data size: 5445 is greater than 2000 bytes, is compressed: true, ZNRecord.id: record. Data will not be written to Zookeeper.
2021-02-07T00:51:59.3116568Z 12175 [main] ERROR org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks  - Failed to handle unknown return code 255. Skip retrying.
2021-02-07T00:51:59.3118353Z java.lang.NullPointerException
2021-02-07T00:51:59.3120109Z 	at org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks$DefaultCallback.needRetry(ZkAsyncCallbacks.java:257)
2021-02-07T00:51:59.3123183Z 	at org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks$DefaultCallback.callback(ZkAsyncCallbacks.java:179)
2021-02-07T00:51:59.3125884Z 	at org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks$CreateCallbackHandler.processResult(ZkAsyncCallbacks.java:104)
2021-02-07T00:51:59.3128265Z 	at org.apache.helix.zookeeper.zkclient.ZkClient.doAsyncCreate(ZkClient.java:1958)
2021-02-07T00:51:59.3130143Z 	at org.apache.helix.zookeeper.zkclient.ZkClient.asyncCreate(ZkClient.java:1937)
2021-02-07T00:51:59.3133079Z 	at org.apache.helix.zookeeper.impl.client.TestRawZkClient.testAsyncWriteByExpectedSession(TestRawZkClient.java:881)
2021-02-07T00:51:59.3136891Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-02-07T00:51:59.3139471Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-02-07T00:51:59.3141856Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-02-07T00:51:59.3143274Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2021-02-07T00:51:59.3144771Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:76)
2021-02-07T00:51:59.3146408Z 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:673)
2021-02-07T00:51:59.3147633Z 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:846)
2021-02-07T00:51:59.3148972Z 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1170)
2021-02-07T00:51:59.3150621Z 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
2021-02-07T00:51:59.3152381Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
2021-02-07T00:51:59.3153490Z 	at org.testng.TestRunner.runWorkers(TestRunner.java:1147)
2021-02-07T00:51:59.3154446Z 	at org.testng.TestRunner.privateRun(TestRunner.java:749)
2021-02-07T00:51:59.3155287Z 	at org.testng.TestRunner.run(TestRunner.java:600)
2021-02-07T00:51:59.3156131Z 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:317)
2021-02-07T00:51:59.3157142Z 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:312)
2021-02-07T00:51:59.3159928Z 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:274)
2021-02-07T00:51:59.3161092Z 	at org.testng.SuiteRunner.run(SuiteRunner.java:223)
2021-02-07T00:51:59.3167789Z 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
2021-02-07T00:51:59.3169224Z 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
2021-02-07T00:51:59.3170553Z 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1039)
2021-02-07T00:51:59.3172114Z 	at org.testng.TestNG.runSuitesLocally(TestNG.java:964)
2021-02-07T00:51:59.3173062Z 	at org.testng.TestNG.run(TestNG.java:900)
2021-02-07T00:51:59.3205773Z 	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
2021-02-07T00:51:59.3208712Z 	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
2021-02-07T00:51:59.3211191Z 	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
2021-02-07T00:51:59.3213740Z 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
2021-02-07T00:51:59.3216525Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
2021-02-07T00:51:59.3218481Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
2021-02-07T00:51:59.3220125Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
2021-02-07T00:51:59.3299426Z 12193 [main] ERROR org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks  - Failed to handle unknown return code 255. Skip retrying.
2021-02-07T00:51:59.3301884Z java.lang.NullPointerException
2021-02-07T00:51:59.3303954Z 	at org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks$DefaultCallback.needRetry(ZkAsyncCallbacks.java:257)
2021-02-07T00:51:59.3306471Z 	at org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks$DefaultCallback.callback(ZkAsyncCallbacks.java:179)
2021-02-07T00:51:59.3309400Z 	at org.apache.helix.zookeeper.zkclient.callback.ZkAsyncCallbacks$SetDataCallbackHandler.processResult(ZkAsyncCallbacks.java:71)
2021-02-07T00:51:59.3312024Z 	at org.apache.helix.zookeeper.zkclient.ZkClient.doAsyncSetData(ZkClient.java:1996)
2021-02-07T00:51:59.3315123Z 	at org.apache.helix.zookeeper.zkclient.ZkClient.asyncSetData(ZkClient.java:1976)
2021-02-07T00:51:59.3319179Z 	at org.apache.helix.zookeeper.impl.client.TestRawZkClient.testAsyncWriteByExpectedSession(TestRawZkClient.java:909)
2021-02-07T00:51:59.3322774Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-02-07T00:51:59.3324262Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-02-07T00:51:59.3326589Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-02-07T00:51:59.3327996Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2021-02-07T00:51:59.3329862Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:76)
2021-02-07T00:51:59.3331727Z 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:673)
2021-02-07T00:51:59.3333151Z 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:846)
2021-02-07T00:51:59.3335123Z 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1170)
2021-02-07T00:51:59.3337194Z 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
2021-02-07T00:51:59.3339200Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
2021-02-07T00:51:59.3340542Z 	at org.testng.TestRunner.runWorkers(TestRunner.java:1147)
2021-02-07T00:51:59.3341670Z 	at org.testng.TestRunner.privateRun(TestRunner.java:749)
2021-02-07T00:51:59.3342553Z 	at org.testng.TestRunner.run(TestRunner.java:600)
2021-02-07T00:51:59.3343746Z 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:317)
2021-02-07T00:51:59.3344970Z 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:312)
2021-02-07T00:51:59.3346091Z 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:274)
2021-02-07T00:51:59.3347026Z 	at org.testng.SuiteRunner.run(SuiteRunner.java:223)
2021-02-07T00:51:59.3348415Z 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
2021-02-07T00:51:59.3350035Z 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
2021-02-07T00:51:59.3351208Z 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1039)
2021-02-07T00:51:59.3352630Z 	at org.testng.TestNG.runSuitesLocally(TestNG.java:964)
2021-02-07T00:51:59.3353756Z 	at org.testng.TestNG.run(TestNG.java:900)
2021-02-07T00:51:59.3355162Z 	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
2021-02-07T00:51:59.3357115Z 	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
2021-02-07T00:51:59.3359457Z 	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
2021-02-07T00:51:59.3362231Z 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
2021-02-07T00:51:59.3365116Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
2021-02-07T00:51:59.3418152Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
2021-02-07T00:51:59.3419808Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
2021-02-07T00:51:59.3701372Z 12238 [main] ERROR org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer  - Data size: 8821 is greater than 2000 bytes, is compressed: true, ZNRecord.id: Oversize. Data will not be written to Zookeeper.
2021-02-07T00:51:59.3912869Z 12259 [main] ERROR org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer  - Data size: 8821 is greater than 2000 bytes, is compressed: true, ZNRecord.id: Oversize. Data will not be written to Zookeeper.
2021-02-07T00:52:01.0575165Z Shut down zookeeper at port 2127 in thread TestNGInvoker-testConnectionLossWhileCreateEphemeral()
2021-02-07T00:52:06.0907051Z Start test: testGetChildrenOnLargeNumChildren
2021-02-07T00:52:14.1905327Z 27057 [TestNGInvoker-testGetChildrenOnLargeNumChildren()] ERROR org.apache.helix.zookeeper.zkclient.ZkClient  - Failed to get children for path /testGetChildrenOnLargeNumChildren because of connection loss. Number of children 110000 exceeds limit 100000, aborting retry.
2021-02-07T00:52:16.1198442Z End test: testGetChildrenOnLargeNumChildren
2021-02-07T00:52:16.1800752Z Shut down zookeeper at port 2127 in thread TestNGInvoker-testRetryUntilConnectedAfterConnectionLoss()
2021-02-07T00:52:16.1814901Z Keep creation thread retrying to connect for 10 seconds...
2021-02-07T00:52:16.1819849Z Trying to create ephemeral node...
2021-02-07T00:52:26.1818588Z Restarting zk server...
2021-02-07T00:52:28.6036094Z Ephemeral node created.
2021-02-07T00:52:28.6126685Z old sessionId= 72057607824015362
2021-02-07T00:52:28.6270891Z In New connection In process event:WatchedEvent state:SyncConnected type:None path:null
2021-02-07T00:52:28.7273670Z In Old connection New state Disconnected
2021-02-07T00:52:30.4713493Z In Old connection New state SyncConnected
2021-02-07T00:52:30.5715303Z In New connection In process event:WatchedEvent state:Disconnected type:None path:null
2021-02-07T00:52:31.6260892Z New sessionId= 72057607824015362
2021-02-07T00:52:32.3829750Z In New connection In process event:WatchedEvent state:SyncConnected type:None path:null
2021-02-07T00:52:32.4821464Z In Old connection New state Disconnected
2021-02-07T00:52:33.5652529Z In Old connection New state SyncConnected
2021-02-07T00:52:33.6650737Z In New connection In process event:WatchedEvent state:Disconnected type:None path:null
2021-02-07T00:52:35.5176153Z In New connection In process event:WatchedEvent state:SyncConnected type:None path:null
2021-02-07T00:52:35.6174389Z In Old connection New state Disconnected
2021-02-07T00:52:37.1116511Z In Old connection New state Expired
2021-02-07T00:52:37.1156360Z In Old connection New state SyncConnected
2021-02-07T00:52:37.1163282Z In Old connection New session
2021-02-07T00:52:45.5189057Z After session expiry sessionId= 72057609289859073
2021-02-07T00:52:45.5290612Z Shut down zookeeper at port 2127 in thread main
2021-02-07T00:52:51.0721065Z Starting ZK server at localhost:2358
2021-02-07T00:52:51.4103261Z [ERROR] Tests run: 53, Failures: 1, Errors: 0, Skipped: 4, Time elapsed: 66.545 s <<< FAILURE! - in TestSuite
2021-02-07T00:52:51.4115517Z [ERROR] testAsyncRetryCategories(org.apache.helix.zookeeper.impl.client.TestZkClientAsyncRetry)  Time elapsed: 2.474 s  <<< FAILURE!
2021-02-07T00:52:51.4121734Z java.lang.AssertionError: expected:<true> but was:<false>
2021-02-07T00:52:51.4132941Z 	at org.apache.helix.zookeeper.impl.client.TestZkClientAsyncRetry.testAsyncRetryCategories(TestZkClientAsyncRetry.java:101)
2021-02-07T00:52:51.4138617Z 
2021-02-07T00:52:51.7438726Z [ERROR] Failures: 
2021-02-07T00:52:51.7440815Z [ERROR]   TestZkClientAsyncRetry.testAsyncRetryCategories:101 expected:<true> but was:<false>
2021-02-07T00:52:51.7442238Z [ERROR] Tests run: 53, Failures: 1, Errors: 0, Skipped: 4
2021-02-07T00:52:51.7538620Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project zookeeper-api: There are test failures.
2021-02-07T00:52:51.7542570Z [ERROR] 
2021-02-07T00:52:51.7553166Z [ERROR] Please refer to /home/runner/work/helix/helix/zookeeper-api/target/surefire-reports for the individual test results.
2021-02-07T00:52:51.7562898Z [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
2021-02-07T00:52:51.7566791Z [ERROR] -> [Help 1]
2021-02-07T00:52:51.7569422Z [ERROR] 
2021-02-07T00:52:51.7589669Z [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2021-02-07T00:52:51.7597378Z [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2021-02-07T00:52:51.7600135Z [ERROR] 
2021-02-07T00:52:51.7608530Z [ERROR] For more information about the errors and possible solutions, please read the following articles:
2021-02-07T00:52:51.7617286Z [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
2021-02-07T00:52:51.7933243Z ##[error]Process completed with exit code 1.

--- a/zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestZkClientAsyncRetry.java
+++ b/zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestZkClientAsyncRetry.java
@@ -43,6 +43,9 @@ import static org.apache.zookeeper.KeeperException.Code.CONNECTIONLOSS;
 public class TestZkClientAsyncRetry extends ZkTestBase {
   private final String TEST_ROOT = String.format("/%s", getClass().getSimpleName());
   private final String NODE_PATH = TEST_ROOT + "/async";
+  // Retry wait time is set to 3 times of the retry interval so as to leave extra buffer in case
+  // the test environment is slow. Extra wait time won't impact the test logic.
+  private final long RETRY_OPS_WAIT_TIMEOUT_MS = 3 * MockAsyncZkClient.RETRY_INTERVAL_MS;
 
   private org.apache.helix.zookeeper.zkclient.ZkClient _zkClient;
   private String _zkServerAddress;
@@ -98,14 +101,14 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
           // Change the mock response
           testZkClient.setAsyncCallRC(KeeperException.Code.OK.intValue());
           // Async retry will succeed now. Wait until the operation is successfully done and verify.
-          Assert.assertTrue(waitAsyncOperation(createCallback, 1000));
+          Assert.assertTrue(waitAsyncOperation(createCallback, RETRY_OPS_WAIT_TIMEOUT_MS));
           Assert.assertEquals(createCallback.getRc(), KeeperException.Code.OK.intValue());
           Assert.assertTrue(testZkClient.exists(NODE_PATH));
           Assert.assertTrue(testZkClient.getAndResetRetryCount() >= 1);
         } else {
           // Async create will fail due to the mock error rc is not recoverable.
           testZkClient.asyncCreate(NODE_PATH, null, CreateMode.PERSISTENT, createCallback);
-          Assert.assertTrue(waitAsyncOperation(createCallback, 1000));
+          Assert.assertTrue(waitAsyncOperation(createCallback, RETRY_OPS_WAIT_TIMEOUT_MS));
           Assert.assertEquals(createCallback.getRc(), code.intValue());
           Assert.assertEquals(testZkClient.getAndResetRetryCount(), 0);
         }
@@ -141,7 +144,7 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
       // Change the mock return code.
       testZkClient.setAsyncCallRC(KeeperException.Code.OK.intValue());
       // Async retry will succeed now. Wait until the operation is successfully done and verify.
-      Assert.assertTrue(waitAsyncOperation(setCallback, 1000));
+      Assert.assertTrue(waitAsyncOperation(setCallback, RETRY_OPS_WAIT_TIMEOUT_MS));
       Assert.assertEquals(setCallback.getRc(), KeeperException.Code.OK.intValue());
       Assert.assertEquals(((ZNRecord) testZkClient.readData(NODE_PATH)).getSimpleField("test"),
           "data");
@@ -160,7 +163,7 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
       // Change the mock return code.
       testZkClient.setAsyncCallRC(KeeperException.Code.OK.intValue());
       // Async retry will succeed now. Wait until the operation is successfully done and verify.
-      Assert.assertTrue(waitAsyncOperation(deleteCallback, 1000));
+      Assert.assertTrue(waitAsyncOperation(deleteCallback, RETRY_OPS_WAIT_TIMEOUT_MS));
       Assert.assertEquals(deleteCallback.getRc(), KeeperException.Code.OK.intValue());
       Assert.assertFalse(testZkClient.exists(NODE_PATH));
       Assert.assertTrue(testZkClient.getAndResetRetryCount() >= 1);
@@ -197,7 +200,7 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
       // Throw exception in retry
       testZkClient.setZkExceptionInRetry(true);
       // Async retry will succeed now. Wait until the operation is done and verify.
-      Assert.assertTrue(waitAsyncOperation(createCallback, 1000),
+      Assert.assertTrue(waitAsyncOperation(createCallback, RETRY_OPS_WAIT_TIMEOUT_MS),
           "Async callback should have been canceled");
       Assert.assertEquals(createCallback.getRc(), CONNECTIONLOSS.intValue());
       Assert.assertTrue(testZkClient.getAndResetRetryCount() >= 1);
@@ -219,7 +222,7 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
       // Throw exception in retry
       testZkClient.setZkExceptionInRetry(true);
       // Async retry will succeed now. Wait until the operation is done and verify.
-      Assert.assertTrue(waitAsyncOperation(setCallback, 1000),
+      Assert.assertTrue(waitAsyncOperation(setCallback, RETRY_OPS_WAIT_TIMEOUT_MS),
           "Async callback should have been canceled");
       Assert.assertEquals(setCallback.getRc(), CONNECTIONLOSS.intValue());
       Assert.assertTrue(testZkClient.getAndResetRetryCount() >= 1);
@@ -251,7 +254,7 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
       // Change the mock return code.
       testZkClient.setAsyncCallRC(KeeperException.Code.OK.intValue());
       // Async retry will succeed now. Wait until the operation is successfully done and verify.
-      Assert.assertTrue(waitAsyncOperation(existsCallback, 1000));
+      Assert.assertTrue(waitAsyncOperation(existsCallback, RETRY_OPS_WAIT_TIMEOUT_MS));
       Assert.assertEquals(existsCallback.getRc(), KeeperException.Code.OK.intValue());
       Assert.assertTrue(existsCallback._stat != null);
       Assert.assertTrue(testZkClient.getAndResetRetryCount() >= 1);
@@ -269,7 +272,7 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
       // Change the mock return code.
       testZkClient.setAsyncCallRC(KeeperException.Code.OK.intValue());
       // Async retry will succeed now. Wait until the operation is successfully done and verify.
-      Assert.assertTrue(waitAsyncOperation(getCallback, 1000));
+      Assert.assertTrue(waitAsyncOperation(getCallback, RETRY_OPS_WAIT_TIMEOUT_MS));
       Assert.assertEquals(getCallback.getRc(), KeeperException.Code.OK.intValue());
       ZNRecord record = testZkClient.deserialize(getCallback._data, NODE_PATH);
       Assert.assertEquals(record.getSimpleField("foo"), "bar");
@@ -303,7 +306,7 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
         Assert.assertEquals(cb.getRc(), CONNECTIONLOSS.intValue());
       }
       // Wait for a while, no callback finishes
-      Assert.assertFalse(waitAsyncOperation(existsCallbacks[0], 1000));
+      Assert.assertFalse(waitAsyncOperation(existsCallbacks[0], RETRY_OPS_WAIT_TIMEOUT_MS));
       for (ZkAsyncCallbacks.ExistsCallbackHandler cb : existsCallbacks) {
         Assert.assertEquals(cb.getRc(), CONNECTIONLOSS.intValue());
         Assert.assertFalse(cb.isOperationDone());
@@ -311,7 +314,7 @@ public class TestZkClientAsyncRetry extends ZkTestBase {
       testZkClient.close();
       // All callback retry will be cancelled because the zkclient is closed.
       for (ZkAsyncCallbacks.ExistsCallbackHandler cb : existsCallbacks) {
-        Assert.assertTrue(waitAsyncOperation(cb, 1000));
+        Assert.assertTrue(waitAsyncOperation(cb, RETRY_OPS_WAIT_TIMEOUT_MS));
         Assert.assertEquals(cb.getRc(), CONNECTIONLOSS.intValue());
       }
       Assert.assertTrue(testZkClient.getAndResetRetryCount() >= 1);

